packages/bp/src/admin/ui/src/app/routes/SegmentHandler.tsx

```diff
@@ -0,0 +1,88 @@
+import segmentPlugin from '@analytics/segment'
+import Analytics, { AnalyticsInstance, PageData } from 'analytics'
+import hash from 'hash.js'
+import React from 'react'
+import { connect } from 'react-redux'
+import { UserState } from '~/user/reducer'
+
+const APP_NAME = 'ADMIN_ANALYTICS' // for reference, in case of second account
+const WRITE_KEY = 'zE0mct7hGOZRtCyImjX9vT1NJ2TpfyGF' // taken from Segment UI
+
+let analytics: AnalyticsInstance
+let loggedIn: boolean = false
+
+const extractUserHashFromUser = (user: UserState): string | undefined => {
+  if (user?.profile?.email) {
+    return hash
+      .sha256()
+      .update('botpressUserEmail' + user.profile.email)
+      .digest('hex')
+  }
+}
+
+const initSegmentAnalytics = async () => {
+  analytics = Analytics({
+    app: APP_NAME,
+    plugins: [
+      segmentPlugin({
+        writeKey: WRITE_KEY
+      })
+    ]
+  })
+
+  await analytics.page()
+
+  return analytics
+}
+
+const handleNewUserState = (user: UserState): void => {
+  const userEmailHash = extractUserHashFromUser(user)
+
+  if (!loggedIn && userEmailHash) {
+    loggedIn = true
+    // @ts-ignore
+    void analytics.identify(null, {
+      userEmailHash,
+      machineUUID: window.UUID
+    })
+  }
+}
+
+function SegmentHandler(props) {
+  if (window.SEND_USAGE_STATS) {
+    React.useEffect(() => {
+      void (async () => {
+        if (!analytics) {
+          await initSegmentAnalytics()
+        }
+
+        handleNewUserState(props.user)
+      })()
+    }, [props.user])
+  }
```

Comment:
Please be consistent in your code usage, prefer arrow functions when possible. Sometimes react complains when the useEffect is in a condition, I suggest moving the if inside. handleNewUser should be a promise, like initSegment